### PR TITLE
Chose provider when assigning a virtual card

### PR DESCRIPTION
**Pull request by Forster-Groux**
Resolve https://github.com/opencollective/opencollective/issues/4807

Requires API: https://github.com/opencollective/opencollective-api/pull/6731

# Description

An organization can now choose between provider (stripe and privacy) when assigning a virtual card.

# Screenshots

provider

### CODE REVIEW

File: components/edit-collective/sections/HostVirtualCards.js [hunk 1]
```diff
@@ -354,7 +357,8 @@ const HostVirtualCards = props => {
             {...vc}
             onSuccess={refetch}
             editHandler={() => setEditingVirtualCard(vc)}
-            hasActions
+            // Edit card action need to be reworked
```
Comment (SudharakaP) [inline components/edit-collective/sections/HostVirtualCards.js:360]: Maybe add a `TODO` to support better searching in the future, 

```suggestion
            // TODO: Edit card action need to be reworked
```

File: components/edit-collective/VirtualCard.js [hunk 2]
```diff
@@ -177,12 +177,20 @@ ActionsButton.propTypes = {
...
+  const value = formatCurrency(spendingLimitAmount, 'USD');
+  if (!spendingLimitAmount) {
     return <FormattedMessage id="VirtualCards.NoLimit" defaultMessage="No Limit" />;
   }
-  switch (spend_limit_duration) {
+  switch (spendingLimitInterval) {
+    case 'DAILY':
+      return (
+        <Fragment>
+          <FormattedMessage id="VirtualCards.LimitedTo" defaultMessage="Limited to" />
```
Comment (SudharakaP) [inline components/edit-collective/VirtualCard.js:189]: The ids are not required for new strings as per, https://github.com/opencollective/opencollective/issues/4697#issuecomment-921858385

```suggestion
          <FormattedMessage defaultMessage="Limited to" />
```

File: components/edit-collective/VirtualCard.js [hunk 3]
```diff
@@ -177,12 +177,20 @@ ActionsButton.propTypes = {
...
+          &nbsp;
+          {value}/<FormattedMessage id="Frequency.Daily.Short" defaultMessage="day" />
```
Comment (SudharakaP) [inline components/edit-collective/VirtualCard.js:191]:
```suggestion
          {value}/<FormattedMessage defaultMessage="day" />
```

File: components/edit-collective/AssignVirtualCardModal.js [hunk 4]
```diff
@@ -268,6 +286,31 @@ const AssignVirtualCardModal = ({ collective, host, virtualCard, onSuccess, onCl
                 />
               )}
             </StyledInputField>
+
+            <StyledInputField
+              gridColumn="1/3"
+              labelFontSize="13px"
+              label="What provider do you use for this card?"
```
Comment (SudharakaP) [inline components/edit-collective/AssignVirtualCardModal.js:293]: Maybe internationalize this label. 😉

Comment (Forster-Groux) [reply components/edit-collective/AssignVirtualCardModal.js]: This is a separate issue : https://github.com/opencollective/opencollective/issues/4857

Comment (SudharakaP) [reply components/edit-collective/AssignVirtualCardModal.js]: Ops, missed that one; I forgot my own issue. 😉

**Pull request merged**